ui/src/main.mjs

```diff
@@ -71,20 +74,69 @@ fetch('_setup')
 
         store.commit('setup/set', setup)
 
-        const socket = io(setup.socketio)
-
-        // handle final disconnection
-        socket.on('disconnect', (reason) => {
-            console.log('SIO disconnected', reason)
-        })
-
-        socket.on('connect', () => {
-            console.log('SIO connected')
-        })
+        let disconnected = false
+        let disconnectedAt = null
+
+        const MAX_TIMER = 300000 // 5 minutes
+
+        let socket = null
+
+        function connect () {
+            socket = io(setup.socketio)
+
+            // handle final disconnection
+            socket.on('disconnect', (reason) => {
+                disconnected = true
+                disconnectedAt = new Date()
```

Comment:
```suggestion
                if (!disconnected) {
                    disconnectedAt = new Date()
                    disconnected = true
                }
```

Recommendation: prevent `disconnectedAt` being updated if already disconnected. It is an unlikely scenario but in the event of a bug, we wont update the disconnection timestamp and the circuit breaker logic will work as designed.